src/schema.ts

```diff
@@ -228,14 +231,17 @@ function binSummary(maxbins: number, summary: Summary): Summary {
 }
 
 /** @return a modified version of the passed summary with unique and distinct set according to the timeunit */
-function timeSummary(timeunit: TimeUnit, summary: Summary): Summary {
+function timeSummary(timeunit: TimeUnit, summary: Summary, excludeInvalid: boolean): Summary {
   const result = extend({}, summary);
 
   var unique: {[value: string]: number} = {};
   keys(summary.unique).forEach(function(dateString) {
-    let date: Date = new Date(dateString);
-    let key: string = ((timeunit === TimeUnit.DAY) ? date.getDay() : convert(timeunit, new Date(dateString))).toString();
-    unique[key] = unique[key] ? unique[key] + summary.unique[dateString] : summary.unique[dateString];
+    // only add key to unique if excludeInvalid is not set or if it is NOT undefined, null, or NaN
```

Comment:
> only add key to unique if excludeInvalid is not set or if it is NOT undefined, null, or NaN

This will make the cache data incorrect if the first time you ask for `excludeInvalid = true` and the next time you ask for false.  

We discussed earlier today that we should always add these invalid values to unique dictionary but use `unique` as a part of cardinality calculation.  I'm not sure why you change it like this and it seems incorrect.